stl/inc/execution

```diff
@@ -3392,12 +3397,13 @@ _FwdIt _Partition_merge(const _FwdIt _False_first, const _FwdIt _True_first, con
     const _Iter_diff_t<_FwdIt> _Count1, const _Iter_diff_t<_FwdIt> _Count2) {
     // Merge partition ranges where [_False_first, _True_first) are falses, [_True_first, _True_last) are trues
     // pre: _Count1 == distance(_False_first, _True_first) && _Count2 == distance(_True_first, _True_last)
+    _REQUIRE_CPP17_MUTABLE_ITERATOR(_FwdIt);

```

Comment:
`_REQUIRE_MEOW` checks are being added to internal helpers, but I believe that our policy is to perform them at the beginning of user-visible parallel algorithms only, so that they are diagnosed as early as possible.

You reverted a couple of these, but `_Partition_merge` is still affected. I am unsure if this is the last occurrence.